Gradle does not work with Yarn

##### **Overview of the issue**
Gradle projects no longer build when choosing Yarn as the client package manager.

```
D:\dev\code\temp\ngx-gradle-fr>gradlew
Starting a Gradle Daemon, 4 stopped Daemons could not be reused, use --status for details

FAILURE: Build failed with an exception.

* Where:
Script 'D:\dev\code\temp\ngx-gradle-fr\gradle\profile_dev.gradle' line: 36

* What went wrong:
A problem occurred evaluating script.
> Could not get unknown property 'yarnInstall' for task ':webpackBuildDev' of type com.moowork.gradle.node.yarn.YarnTask.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s
```

##### **Motivation for or Use Case**
Gradle projects should work with Yarn as well

##### **Reproduce the error**
Create a new project using gradle and yarn. Then do `./gradlew` and build will fail

##### **Related issues**
This is a regression after merging #9389 
<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**
The problem is that there is not gradle task `yarnInstall`. The proper name is `yarn` instead

Will submit a PR to fix this.

##### **JHipster Version(s)**
6./x (Current master)

##### **JHipster configuration**
```
{
  "generator-jhipster": {
    "baseName": "travisNg2Jwt",
    "packageName": "io.github.jhipster.travis",
    "packageFolder": "io/github/jhipster/travis",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "buildTool": "gradle",
    "useSass": true,
    "applicationType": "monolith",
    "testFrameworks": [
      "gatling",
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "fr",
    "languages": [
      "en",
      "fr"
    ],
    "travis": true,
    "serviceDiscoveryType": false,
    "clientPackageManager": "yarn",
    "clientFramework": "angularX",
    "jhipsterVersion": "5.8.2",
    "jwtSecretKey": "MmRkYzk5ZmVlNzQ1NDM1MTA4OTFmZjUyYTA0MDMyZWJmYmY3YjRmMjRjMjI3MWNmZWU1NWJlZjlhYmNlYzY1ODFmYjFjMTQ3MjJjODA2MzA1ZTdiNTJmMzAxZGJhOTQ4NjY5MDg0OGM4NWM2MjBlNWZlZjcwOWNjMTIwZjE2MTg=",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": []
  }
}
```

##### **Browsers and Operating System**
Windows 10 x64

-   [x] Checking this box is mandatory (this is just to show you read everything)

--- a/generators/server/templates/gradle/profile_dev.gradle.ejs
+++ b/generators/server/templates/gradle/profile_dev.gradle.ejs
@@ -58,7 +58,7 @@ task webpackBuildDev(type: <%= _.upperFirst(clientPackageManager) %>Task) {
 
     outputs.dir("<%= CLIENT_DIST_DIR %>")
 
-    dependsOn <%= clientPackageManager %><%_ if (clientPackageManager==='npm') { _%>Install<%_ } _%>
+    dependsOn <%= clientPackageManager %><%_ if (clientPackageManager === 'npm') { _%>Install<%_ } _%>
 
     args = ["run", "webpack:build"]
 }

--- a/generators/server/templates/gradle/profile_dev.gradle.ejs
+++ b/generators/server/templates/gradle/profile_dev.gradle.ejs
@@ -45,7 +45,11 @@ bootRun {
 
 <%_ if (!skipClient) { _%>
 task webpackBuildDev(type: <%= _.upperFirst(clientPackageManager) %>Task) {
+    <%_ if (clientPackageManager==='npm') { _%>
     inputs.file("package-lock.json")
+    <%_ } else { _%>
+    inputs.file("yarn.lock")
+    <%_ } _%>
     inputs.dir("<%= CLIENT_MAIN_SRC_DIR %>")
 
     def webpackDevFiles = fileTree("<%= CLIENT_WEBPACK_DIR %>/")
@@ -54,7 +58,7 @@ task webpackBuildDev(type: <%= _.upperFirst(clientPackageManager) %>Task) {
 
     outputs.dir("<%= CLIENT_DIST_DIR %>")
 
-    dependsOn <%= clientPackageManager %>Install
+    dependsOn <%= clientPackageManager %>_install
 
     args = ["run", "webpack:build"]
 }

--- a/generators/server/templates/gradle/profile_dev.gradle.ejs
+++ b/generators/server/templates/gradle/profile_dev.gradle.ejs
@@ -58,7 +58,7 @@ task webpackBuildDev(type: <%= _.upperFirst(clientPackageManager) %>Task) {
 
     outputs.dir("<%= CLIENT_DIST_DIR %>")
 
-    dependsOn <%= clientPackageManager %>_install
+    dependsOn <%= clientPackageManager %><%_ if (clientPackageManager==='npm') { _%>Install<%_ } _%>
 
     args = ["run", "webpack:build"]
 }

--- a/generators/server/templates/gradle/profile_prod.gradle.ejs
+++ b/generators/server/templates/gradle/profile_prod.gradle.ejs
@@ -40,11 +40,11 @@ bootRun {
 }
 
 <%_ if (!skipClient) { _%>
-task webpack_test(type: <%= _.upperFirst(clientPackageManager) %>Task, dependsOn: "<%= clientPackageManager %><%_ if (clientPackageManager==='npm') { _%>Install<%_ } _%>") {
+task webpack_test(type: <%= _.upperFirst(clientPackageManager) %>Task, dependsOn: "<%= clientPackageManager %><%_ if (clientPackageManager === 'npm') { _%>Install<%_ } _%>") {
     args = ["run", "webpack:test"]
 }
 
-task webpack(type: <%= _.upperFirst(clientPackageManager) %>Task, dependsOn: "<%= clientPackageManager %><%_ if (clientPackageManager==='npm') { _%>Install<%_ } _%>") {
+task webpack(type: <%= _.upperFirst(clientPackageManager) %>Task, dependsOn: "<%= clientPackageManager %><%_ if (clientPackageManager === 'npm') { _%>Install<%_ } _%>") {
     args = ["run", "webpack:prod"]
 }
 <%_ } _%>

--- a/generators/server/templates/gradle/profile_prod.gradle.ejs
+++ b/generators/server/templates/gradle/profile_prod.gradle.ejs
@@ -40,11 +40,11 @@ bootRun {
 }
 
 <%_ if (!skipClient) { _%>
-task webpack_test(type: <%= _.upperFirst(clientPackageManager) %>Task, dependsOn: "<%= clientPackageManager %>Install") {
+task webpack_test(type: <%= _.upperFirst(clientPackageManager) %>Task, dependsOn: "<%= clientPackageManager %><%_ if (clientPackageManager==='npm') { _%>Install<%_ } _%>") {
     args = ["run", "webpack:test"]
 }
 
-task webpack(type: <%= _.upperFirst(clientPackageManager) %>Task, dependsOn: "<%= clientPackageManager %>Install") {
+task webpack(type: <%= _.upperFirst(clientPackageManager) %>Task, dependsOn: "<%= clientPackageManager %><%_ if (clientPackageManager==='npm') { _%>Install<%_ } _%>") {
     args = ["run", "webpack:prod"]
 }
 <%_ } _%>

